_posts/2015-03-16-CocoaPods.org-Take-Two.md

```diff
@@ -0,0 +1,151 @@
+  {% breaking_image http://cocoadocs.org/docsets/MiniFuture/0.1.0/preview.png %}
+  
+* We then generate an estimated quality number for the library. This is based on a collection of [individual metrics][7] that are applied to the generated metrics data. They take a library and add or remove points based whether the rule applies for that library. These range from the popularity of the library, total test expectations / lines of code to the files, and average lines of code per file in the library.
+
+```json
+// https://cocoadocs-api-cocoapods-org.herokuapp.com/pods/MiniFuture/stats
+
+[
+	{
+	    "applies_for_pod": false,
+	    "description": "Testing a library shows that the developers care about long term quality on a project as internalized logic is made explicit via testing.",
+	    "modifier": -20,
+	    "title": "Test Expectations / Line of Code"
+	},
+	{
+	    "applies_for_pod": false,
+	    "description": "Too big of a library can impact startup time, and add redundant assets.",
+	    "modifier": -10,
+	    "title": "Download size"
+	},
+	{
+	    "applies_for_pod": false,
+	    "description": "Smaller, more composeable classes tend to be easier to understand.",
+	    "modifier": -8,
+	    "title": "Lines of Code / File"
+	},
+	[...]
+	{
+	    "applies_for_pod": true,
+	    "description": "A popular library means there can be a community to help improve and maintain a project.",
+	    "modifier": 5,
+	    "title": "Is popular"
+	}
+]
+```
+
+This gives enough metadata to start creating a page that represents the library with the kind of context that is specific to our community. In a lot of ways it can be better than the front page of the GitHub repo because of this.
+
+## How it comes together
+
+We use the quality metrics in search, this is used to order the results we get back. There are quite a few different [sorting options](https://github.com/CocoaPods/search.cocoapods.org/blob/238b497872857e7b44ac715170888ef7f5ba046b/lib/search.rb#L345) now, so we may look into offering choices there.
+
+We've started exposing a good chunk of the information in a pod's profile pages, and via inline results but not all of it. There's [interesting](https://github.com/CocoaPods/cocoapods.org/issues/107) [issues](https://github.com/CocoaPods/cocoapods.org/issues/106) around how we can expose more.
+
+## Going forward
+
+The APIs we use are all available for anyone to build things on top of. This data is pretty solid, the API routes won't be changing and work is on-going to make it more accurate. We'll only be expanding from here.
+
+It's not perfect, but a website isn't an app and we can deploy daily. We re-wrote the homepage to run directly from the new databases and it's simplified the development process a lot. So it's easy for anyone to contribute and help out. We'll be iterating on it for a while before stablizing, so we'd love feedback in [an issue](https://github.com/CocoaPods/cocoapods.org/issues/new) or on twitter.
```

Comment:
web apps are apps though (I get what you're saying, it's not like an iOS app that needs to be packaged and cannot be deployed continuously) but technically websites are a different kind of app.